uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -562,6 +695,85 @@ public function testLeaveRequestBalanceForEntitlementWithParamsToBothExcludeAndI
   }
 
+  public function testLeaveRequestBalanceForEntitlementShouldIncludeTOILBalanceChanges() {
+    $entitlement = $this->createLeavePeriodEntitlementMockForBalanceTests(
+      new DateTime('today'),
+      new DateTime('+100 days')
+    );
+    
+    $balance = LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($entitlement);
+    $this->assertEquals(0, $balance);
+
+    TOILRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $entitlement->contact_id,
+      'type_id' => $entitlement->type_id,
+      'from_date' => CRM_Utils_Date::processDate('today'),
+      'to_date' => CRM_Utils_Date::processDate('today'),
+      'duration' => 360,
+      'toil_to_accrue' => 2,
+      'expiry_date' => CRM_Utils_Date::processDate('+30 days'),
+    ]);
+
+    // This will deduct 1 day from the 2 accrued by the toil request
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $entitlement->contact_id,
+      'type_id' => $entitlement->type_id,
+      'from_date' => CRM_Utils_Date::processDate('+10 days'),
+      'to_date' => CRM_Utils_Date::processDate('+10 days')
+    ], true);
+
+
+    $balance = LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($entitlement);
+    $this->assertEquals(1, $balance);
+  }
+
+  public function testLeaveRequestBalanceForEntitlementDoesNotIncludeExpiredTOILBalanceChangeByDefault() {
+    $entitlement = $this->createLeavePeriodEntitlementMockForBalanceTests(
+      new DateTime('today'),
+      new DateTime('+100 days')
+    );
+
+    $balance = LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($entitlement);
+    $this->assertEquals(0, $balance);
+
+    $toilRequest = TOILRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $entitlement->contact_id,
+      'type_id' => $entitlement->type_id,
+      'from_date' => CRM_Utils_Date::processDate('today'),
+      'to_date' => CRM_Utils_Date::processDate('today'),
+      'duration' => 360,
+      'toil_to_accrue' => 2,
+      'expiry_date' => CRM_Utils_Date::processDate('+30 days'),
+    ]);
+
+    // This will deduct 1 day from the 2 accrued by the toil request
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $entitlement->contact_id,
+      'type_id' => $entitlement->type_id,
+      'from_date' => CRM_Utils_Date::processDate('+10 days'),
+      'to_date' => CRM_Utils_Date::processDate('+10 days')
+    ], true);
+
+    $toilBalanceChange = $this->findToilRequestBalanceChange($toilRequest->id);
+
+    //This balance change should not be included on the count
+    LeaveBalanceChangeFabricator::fabricate([
+      'source_id' => $toilBalanceChange->source_id,
+      'source_type' => $toilBalanceChange->source_type,
+      'expiry_date' => CRM_Utils_Date::processDate($toilBalanceChange->expiry_date),
+      'type_id' => $toilBalanceChange->type_id,
+      'amount' => -1,
+      'expired_balance_change_id' => $toilBalanceChange->id
+    ]);
+
```

Comment:
Why can't the `createExpiredTOILRequestBalanceChange` in LeaveBalanceChangeHelpersTrait.php be used here?